docs/document/content/features/sharding/principle/parse.en.md

```diff
@@ -25,10 +25,56 @@ At last, through traversing the abstract syntax tree, the context needed by shar
 
-The third generation of SQL parsing engine begins from 3.0.x version. ShardingSphere tries to adopts ANTLR as the SQL parsing engine, and uses Visit to obtain SQL Statement from AST. Starting from version 5.0.x, the architecture of the parsing engine has been refactored. At the same time, it is convenient to directly obtain the parsing results of the same SQL to improve parsing efficiency by putting the AST obtained from the first parsing into the cache. Therefore, we recommend that users adopt `PreparedStatement` this SQL pre-compilation method to improve performance. Currently, users can also use ShardingSphere's SQL parsing engine independently to obtain AST and SQL Statements for a variety of mainstream relational databases. In the future, the SQL parsing engine will continue to provide powerful functions such as SQL formatting and SQL templating.
+The third generation of SQL parsing engine begins from 3.0.x version. ShardingSphere tries to adopts ANTLR as a generator for the SQL parsing engine, and uses Visit to obtain SQL Statement from AST. Starting from version 5.0.x, the architecture of the parsing engine has been refactored. At the same time, it is convenient to directly obtain the parsing results of the same SQL to improve parsing efficiency by putting the AST obtained from the first parsing into the cache. Therefore, we recommend that users adopt `PreparedStatement` this SQL pre-compilation method to improve performance. Currently, users can also use ShardingSphere's SQL parsing engine independently to obtain AST and SQL Statements for a variety of mainstream relational databases. In the future, the SQL parsing engine will continue to provide powerful functions such as SQL formatting and SQL templating.
+
+### Features
+
+* Independent SQL parsing engine
+* Support multiple dialects
+
+| DB    | Status |
+|----------|--------|
+|MySQL     |supported|
+|PostgreSQL|supported|
+|SQLServer |supported|
+|Oracle    |supported|
+|SQL92     |supported|
+
+* SQL format (developing)
+* SQL parameterize (developing)
+
+### Advantages
+The syntax rules can be easily expanded and modified by using `ANTLR`
+
+### API Usage
+
+Maven config
+```
+<dependency>
+    <groupId>org.apache.shardingsphere</groupId>
+    <artifactId>shardingsphere-sql-parser-engine</artifactId>
+    <version>${project.version}</version>
+</dependency>
+// According to the needs, introduce the parsing module of the specified dialect (take MySQL as an example)
+<dependency>
+    <groupId>org.apache.shardingsphere</groupId>
+    <artifactId>shardingsphere-sql-parser-mysql</artifactId>
+    <version>${project.version}</version>
+</dependency>
+```
+
+demo:
+
+1. Get AST
+
+```
+ParseTree tree = new SQLParserEngine(databaseType).parse(sql, false)
```

Comment:
First, it is expected to give more explanation for all its parameters. Plus, a list of possible values for `databaseType` is needed as well.

Second, another chapter, e.g., `2. Get different visited results` is also necessary. We can provide other parsed results except for `AST`, which is just a basic one. At least, `SQLStatement` may be an option for users currently, and more wonderful features are ongoing, won't we?

Reply:
+1, Look forward to a simpler return value rather than a tree, maybe? From the SkyWalking analysis requirements, https://github.com/apache/skywalking/issues/5838, we need
1. DML type, create/delete/update. Or it is a DDL.
1. A listener mode or a way to iterate the table lists. I am not sure which way do you provide for now. 

The basic idea is, SkyWalking as an APM, doesn't intend to do routing, sharding, or anything similar, we are planning to leverage this to get the basic profile of the executed SQL statements which were captured by the SkyWalking agents.

Also, I remember there is a cache mechanism in the parser engine, do you have an LRU cache to keep memory safe? Or could I provide an implementation for the cache by following your interface?

Reply:
Hi @wu-sheng 
1. Currently, SS will return the concrete SQLStatement, like `SelectStatement`, or `UpdateStatement`, which means you need some of `IF` code clips to get the corresponding SQL types.
2. If you just wanna get a full table list of `DDL` and `DML` SQL, I guess the visitor result, `SQLStatement` (Not a  comprehensive AST) seems suffienct. No doubt, a test is needed in case.
3. Yep, an LRU cache mechanism is embodied in SQL Parser. So, users do not need to do any other extra work.

Reply:
> If you just wanna get a full table list of DDL and DML SQL, I guess the visitor result, SQLStatement (Not a comprehensive AST) seems suffienct. No doubt, a test is needed in case.

I have to iterate this object, right? 

__
2 more questions
1. Whether `SQLParserEngine` is thread-safe? How to use it in the SkyWalking concurrent streaming process?
1. Do you have a benchmark for the benchmark for this parser?

Reply:
Ok, i'll add some explanation for its parameters. 
1. `SQLParserEngine` is thread-safe.
2. There is no benchmark for this parser for the time being. It will come in the future.